src/interface_adapter/room_settings/RoomSettingsPresenter.java

```diff
@@ -0,0 +1,44 @@
+package interface_adapter.room_settings;
+
+import entities.rooms.Room;
+import interface_adapter.ViewManagerModel;
+import interface_adapter.rooms.RoomsState;
+import interface_adapter.rooms.RoomsViewModel;
+import use_case.room_settings.RoomSettingsOutputBoundary;
+import view.LoggedInView;
+
+public class RoomSettingsPresenter implements RoomSettingsOutputBoundary {
+  private final RoomsViewModel roomsViewModel;
+  private final ViewManagerModel viewManagerModel;
+
+  public RoomSettingsPresenter(RoomsViewModel roomsViewModel, ViewManagerModel viewManagerModel) {
+    this.roomsViewModel = roomsViewModel;
+    this.viewManagerModel = viewManagerModel;
+  }
+
+  public void deletedRoom(Room room) {
+    RoomsState state = roomsViewModel.getState();
+    var availableRooms = state.getAvailableRooms();
+    availableRooms.remove(room);
+    state.setRoomUid(null);
+    roomsViewModel.setState(state);
+    roomsViewModel.firePropertyChanged();
+
+    viewManagerModel.setActiveView(LoggedInView.viewName);
+  }
+
+  public void savedRoomName(Room activeRoom, String roomName) {
```

Comment:
There's no error handling in this file either. Also, please stay consistent with the naming conventions throughout the project and call these functions 'prepare__View__'.